packages/big-design/src/components/AccordionGroup/hooks/useAccordion.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+interface UseAccordionProps {
+  defaultExpanded?: boolean;
+}
+
+export const useAccordion = ({ defaultExpanded }: UseAccordionProps) => {
+  const [isExpanded, setisExpanded] = useState(false);
+
+  useEffect(() => {
+    if (defaultExpanded) {
+      setisExpanded(defaultExpanded);
+    }
+  }, [defaultExpanded]);
+
+  const onClick = () => {
+    const nextIsExpanded = !isExpanded;
+
+    setisExpanded(nextIsExpanded);
```

Comment:
`setIsExpanded(!nextIsExpanded)` 🍹 ?

Reply:
@chanceaclark I see you thumbs down this comment. Can you expand more about the reason why not?

Reply:
See https://github.com/bigcommerce/big-design/pull/933#discussion_r936046033

Reply:
@chanceaclark I don't think an updater function helps here. It's not being called multiple times and this is a click handler, which will get automatic batches even in React < 18.

Reply:
🤦‍♂️ I retract my 👎 

Either way should be fine 🤔